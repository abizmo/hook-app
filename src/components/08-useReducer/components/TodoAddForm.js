import React from 'react'
import PropTypes from 'prop-types'

import useForm from '../../../hooks/useForm'

const TodoAddForm = ({ onAddTodo }) => {
  const [ {todo}, handleTodoChange, reset ] = useForm({ todo: '' })

  const handleSubmit = (evt) => {
    evt.preventDefault();

    if (todo.trim().length > 1) {
      onAddTodo(todo)
      reset()
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <div className="input-group mb-3">
        <input
          onChange={handleTodoChange}
          value={todo}
          type="text"
          className="form-control"
          name="todo"
          placeholder="New Todo"
          aria-label="New Todo"
          aria-describedby="button-add"
        />
        <button
          className="btn btn-primary"
          type="submit"
          id="button-add"
        >Add</button>
      </div>
    </form>
  )
}

TodoAddForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
}

export default TodoAddForm
