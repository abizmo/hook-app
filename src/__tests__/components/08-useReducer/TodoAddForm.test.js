import { shallow } from 'enzyme'

import TodoAddForm from "../../../components/08-useReducer/components/TodoAddForm"

describe('TodoAddForm tests', () => {
  const handleAddTodo = jest.fn()
  const wrapper = shallow(
    <TodoAddForm
      onAddTodo={handleAddTodo}
    />
  )

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button').text().trim()).toBe('Add')
    expect(wrapper.find('input').exists()).toBeTruthy()
  })
  
  test('should no call onAddTodo when no input', () => {
    wrapper.find('form').prop('onSubmit')({
      preventDefault(){}
    })

    expect(handleAddTodo).not.toHaveBeenCalled()
  })

  test('should call onAddTodo when input', () => {
    const value = 'Ir al super'
    wrapper.find('input').simulate('change', {
      target: {
        name: 'todo',
        value
      }
    })
    wrapper.find('form').prop('onSubmit')({
      preventDefault(){}
    })

    expect(handleAddTodo).toHaveBeenCalledWith(value)
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})
