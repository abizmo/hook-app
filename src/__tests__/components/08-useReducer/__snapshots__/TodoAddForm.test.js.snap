// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAddForm tests should render and match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-group mb-3"
  >
    <input
      aria-describedby="button-add"
      aria-label="New Todo"
      className="form-control"
      name="todo"
      onChange={[Function]}
      placeholder="New Todo"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      id="button-add"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
